File: apps/research-front-vue/src/repositories/ArticleRepository.js
import Repository from "./Repository"

const resource = "/articles"

export default {
  get() {
    return Repository.get(`${resource}`)
  },
  getArticle(articleId) {
    return Repository.get(`${resource}/${articleId}/content`)
  },
  getArticlesOverview() {
    return Repository.get(`${resource}`+"?size=10")
  }
}

File: apps/research-api/usecase/port/repository/article_repository.go
package repository

import "research-api/domain"

type ArticleRepository interface {
	FindById(id domain.ArticleId) (domain.Article, error)
	FindLatest(size int) (domain.Articles, error)
	StoreArticle(article domain.Article) (int, error)
	StoreEditor(editor domain.Editor) (domain.Editor, error)
}
